### PR TITLE
Feature/emboss support ueb

### PR DESCRIPTION
**Pull request by jtreuting**
Adding support for the new Braille Type codes when embossing.

- We have to support the old and the new codes right now during the transition
- The logic for finding the attachment is a bit confusing but I added some comments that hopefully are enough to describe the logic, so I won't try to add more here as a way to see if it is descriptive enough

### CODE REVIEW

File: TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java [hunk 1]
```diff
@@ -166,50 +166,60 @@ public void setQti (ITSQTI _qti) {
...
+    // the last part of the code is the braille type
+    //  instead of listing them all, use that to allow for new types to be created without altering this case statement
+    //  those listed above are legacy
+    if (brailleType == null) {
+      String[] parts = accProps.getBrailleType ().split("_");
+      brailleType = parts[parts.length - 1].toLowerCase();
+    }
 
-    final String brailleTypeFinal = brailleType;
+    List<String> brailleSubTypes = new ArrayList<String>();
```
Comment (millerts) [inline TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java]: Could combine this line with `List<String> brailleSubTypes = Arrays.asList(brailleType);`

File: TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java [hunk 2]
```diff
@@ -166,50 +166,60 @@ public void setQti (ITSQTI _qti) {
...
-    // a braille type as well)
-    ITSAttachment mainBrailleAttachment = (ITSAttachment) CollectionUtils.find (_attachments, new Predicate ()
-    {
-      @Override
-      public boolean evaluate (Object attachment) {
-        return StringUtils.equalsIgnoreCase (brailleTypeFinal, ((ITSAttachment) attachment).getSubType ());
-      }
-    });
+    // The ending letter determines the Math encoding
+    //  N = Nemeth, T = UEB Math, L = No Math Code Needed
```
Comment (millerts) [inline TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java]: Is there a good object or place where we could make these magic values constants?

File: TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java [hunk 3]
```diff
@@ -219,6 +229,17 @@ public boolean evaluate(Object attachment) {
   public List<Element> getGenericElements () {
     return _genericElements;
   }
+
+  private ITSAttachment findBrailleAttachment(String brailleSubTypeCode) {
```
Comment (millerts) [inline TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java]: What if this method took in a List of brailSubTypeCodes and does the iterating that you're doing on line 203-209 and 216-222?

File: TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java (hunk 2 above)
Comment (jtreuting) [inline TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java]: This is the only place where they are needed so I wasn't worried about them living here

File: TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java (hunk 1 above)
Comment (jtreuting) [inline TDS.ItemRenderer/src/main/java/tds/itemrenderer/data/ITSContent.java]: Turns out Arrays.asList is immutable so can't add to it later if needed.

**Comment by jtreuting** [comment]
@millerts  Lost one of your comments because of the new commit I guess.  But I combined the logic into the find method

**Pull request merged**